scripts/ingest_GWAS_Catalog.py

```diff
@@ -66,25 +80,136 @@
+    # 'SUBMISSION DATE': 'submission_date',
+    # 'STATISTICAL MODEL': 'statistical_model',
+    'BACKGROUND TRAIT': 'background_trait',
+    'MAPPED BACKGROUND TRAIT': 'mapped_background_trait',
+    'MAPPED BACKGROUND TRAIT URI': 'mapped_background_trait_uri',
+}
+
+def read_study_table(read_unpublished: bool = False) -> pyspark.sql.DataFrame:
+    """
+    This function reads the study table and returns a Spark DataFrame with the columns renamed to match the names
+    in the `STUDY_COLUMNS_MAP` dictionary
+
+    Args:
+      read_unpublished (bool): bool=False. Defaults to False. Indicating if unpublished studies should be read.
+
+    Returns:
+      A dataframe with the columns specified in STUDY_COLUMNS_MAP.
+    """
+
+    # Reading the study table:
+    study_table = spark.read.csv(STUDY_FILE, sep='\t', header=True)
+
+    if read_unpublished:
+
+        # Reading the unpublished study table:
+        unpublished_study_table = (
+            spark.read.csv(UNPUBLISHED_STUDY_FILE, sep='\t', header=True)
+
+            # The column names are expected to be the same in both tables, so this should be gone at some point:
+            .withColumnRenamed('MAPPED TRAIT', 'MAPPED_TRAIT')
+            .withColumnRenamed('MAPPED TRAIT URI', 'MAPPED_TRAIT_URI')
+        )
+
+        # Expression to replace "not yet cureated" string with nulls:
+        expressions = map(
+            lambda column: (column, F.when(F.col(column) == 'not yet curated', F.lit(None)).otherwise(F.col(column))),
+            STUDY_COLUMNS_MAP.keys()
+        )
+        unpublished_study_table = reduce(lambda DF, value: DF.withColumn(*value), expressions, unpublished_study_table)
+
+        study_table = study_table.union(unpublished_study_table)
+
+    # Selecting and renaming relevant columns:
+    return (
+        study_table
+        .select(*[F.col(old_name).alias(new_name) for old_name, new_name in STUDY_COLUMNS_MAP.items()])
+        .persist()
+    )
+
+
+def extract_sample_sizes(df: pyspark.sql.DataFrame) -> pyspark.sql.DataFrame:
+    """
+    It takes a dataframe with a column called `initial_sample_size` and returns a dataframe with columns
+    `n_cases`, `n_controls`, and `n_samples`
+
+    Args:
+      df (pyspark.sql.DataFrame): pyspark.sql.DataFrame
+
+    Returns:
+      A dataframe with the columns:
+        - n_cases
+        - n_controls
+        - n_samples
+    """
+
+    columns = df.columns
+
+    return (
+        df
+        # .withColumn('samples', F.explode_outer(F.col('initial_sample_size')))
+        .withColumn('samples', F.explode(F.split(F.col('initial_sample_size'), r',\s+')))
+
+        # Extracting the sample size from the string:
+        .withColumn('sample_size', F.regexp_extract(F.regexp_replace(F.col('samples'), ',', ''), r'[0-9,]+', 0).cast(T.IntegerType()))
+
+        # Extracting number of cases:
+        .withColumn('n_cases', F.when(F.col('samples').contains('cases'), F.col('sample_size')).otherwise(F.lit(0)))
```

Comment:
If you set this to 0 you might need to update also the lines attached below.

Basically if the n_cases is not null it compute the beta as log(beta) assuming that is odd ratio.

https://github.com/opentargets/genetics-v2d-data/blob/4b5e6f9989dca73f93dcdeaa8fc8d1bd3f51690f/scripts/format_sumstat_toploci_assoc.py#L29

https://github.com/opentargets/genetics-v2d-data/blob/4b5e6f9989dca73f93dcdeaa8fc8d1bd3f51690f/scripts/format_sumstat_toploci_assoc.py#L51

https://github.com/opentargets/genetics-v2d-data/blob/4b5e6f9989dca73f93dcdeaa8fc8d1bd3f51690f/scripts/format_sumstat_toploci_assoc.py#L95

Reply:
You are right. So you mean, when the cases are non-zero, but the controls are zero, I should resolve the cases to zero? (I think conceptually that would be correct) However the code, which processes these values should be ready to handle edge cases.

Reply:
Yes I think when one of them is 0 the other should be too. 

After this we should probably also change the "is_beta = pd.isnull(row['case_prop'])" concept ?!

I can think of a way to handle that